Algorithm.CSharp/AdjustedVolumeRegressionAlgorithm.cs

```diff
@@ -0,0 +1,194 @@
+        private readonly IEnumerator<decimal> _expectedAdjustedAskSize = new List<decimal> { 53900.05m, 1400.00m, 6300.01m, 2100.00m, 1400.00m, 1400.00m, 700.00m, 
+            2100.00m, 3500.00m, 700.00m }.GetEnumerator();
+        private readonly IEnumerator<decimal> _expectedAdjustedBidSize = new List<decimal> { 700.00m, 2800.00m, 700.00m, 700.00m, 700.00m, 1400.00m, 2800.00m,
+            2100.00m, 7700.01m, 700.00m }.GetEnumerator();
+
+        public override void Initialize()
+        {
+            SetStartDate(2014, 6, 5);      //Set Start Date
+            SetEndDate(2014, 6, 5);         //Set End Date
+
+            UniverseSettings.DataNormalizationMode = DataNormalizationMode.SplitAdjusted;
+            _aapl = AddEquity(Ticker, Resolution.Minute).Symbol;
+        }
+
+        /// <summary>
+        /// OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
+        /// </summary>
+        /// <param name="data">Slice object keyed by symbol containing the stock data</param>
+        public override void OnData(Slice data)
+        {
+            if (!Portfolio.Invested)
+            {
+                SetHoldings(_aapl, 1);
+            }
+
+            if (data.Splits.ContainsKey(_aapl))
+            {
+                Log(data.Splits[_aapl].ToString());
+            }
+
+            if (data.Bars.ContainsKey(_aapl))
+            {
+                var aaplData = data.Bars[_aapl];
+                var aaplVolume = decimal.Round(aaplData.Volume, 2);
+
+                // Assert our volume matches what we expect
+                if (_expectedAdjustedVolume.MoveNext() && _expectedAdjustedVolume.Current != aaplVolume)
+                {
+                    // Our values don't match lets try and give a reason why
+                    var dayFactor = _factorFile.GetSplitFactor(aaplData.Time);
+                    var probableAdjustedVolume = decimal.Round(aaplData.Volume / dayFactor, 2);
+
+                    if (_expectedAdjustedVolume.Current == probableAdjustedVolume)
+                    {
+                        throw new ArgumentException($"Volume was incorrect; but manually adjusted value is correct." + 
+                            $" Adjustment by multiplying volume by {1 / dayFactor} is not occurring.");
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"Volume was incorrect; even when adjusted manually by" + 
+                            $" multiplying volume by {1 / dayFactor}. Data may have changed.");
+                    }
+                }
+            }
+
+            if (data.QuoteBars.ContainsKey(_aapl))
+            {
+                var aaplQuoteData = data.QuoteBars[_aapl];
+                var aaplAskSize = decimal.Round(aaplQuoteData.LastAskSize, 2);
+                var aaplBidSize = decimal.Round(aaplQuoteData.LastBidSize, 2);
```

Comment:
hm why do you need to round the volume and sizes here?

Reply:
This is because I rounded my expected values, APPLs factor is like 7.557..... so the end result isn't pretty and just wanted to clean it up for my expected values that are hardcoded 😁